src/main/kotlin/org/amshove/kluent/Collections.kt

```diff
@@ -206,51 +206,51 @@ infix fun Short.shouldBeIn(theArray: ShortArray) = this.apply { this shouldBeIn
 
 infix fun Short.shouldNotBeIn(theArray: ShortArray) = this.apply { this shouldNotBeIn theArray.toTypedArray() }
 
-infix fun <T> Iterable<T>.shouldContain(theThing: T) = this.apply { if (this.contains(theThing)) Unit else fail("$this should contain $theThing", "$theThing", join(this)) }
+infix fun <T, I : Iterable<T>> I.shouldContain(theThing: T) = this.apply { if (this.contains(theThing)) Unit else fail("$this should contain $theThing", "$theThing", join(this)) }
 
-infix fun <T> Iterable<T>.shouldContainSome(things: Iterable<T>) = this.apply { assertTrue("Expected $this to contain at least one of $things", this.any { things.contains(it) }) }
+infix fun <T, I : Iterable<T>> I.shouldContainSome(things: Iterable<T>) = this.apply { assertTrue("Expected $this to contain at least one of $things", this.any { things.contains(it) }) }
```

Comment:
A personal preference: 
* Explicitly specify the return type, so that there is no doubt.
* Omit the `this`

If you don't agree, that is ok. Just my opinion.

Reply:
I agree with the returntype, it's somewhat noted in the new [style guide](https://kotlinlang.org/docs/reference/coding-conventions.html#unit) which I'd like to follow.

It doesn't mention the use of `this` + scope functions within extension methods, but thinking about it I like dropping the first this, because it is harder to understand with `this` also being inside the apply